### PR TITLE
Improve packaging

### PR DESCRIPTION
**Pull request by dweindl**
To maybe fully work on windows...

### CODE REVIEW

File: python/amici/__init__.py [hunk 1]
```diff
@@ -51,12 +55,14 @@
...
-    getDataObservablesAsDataFrame, getSimulationObservablesAsDataFrame, \
-    getSimulationStatesAsDataFrame, getResidualsAsDataFrame
+if has_clibs:
+    # these module require the swig interface
+    from .sbml_import import SbmlImporter, assignmentRules2observables, \
+        constantSpeciesToParameters
+    from .numpy import rdataToNumPyArrays, edataToNumPyArrays
+    from .pandas import getEdataFromDataFrame, \
+        getDataObservablesAsDataFrame, getSimulationObservablesAsDataFrame, \
+        getSimulationStatesAsDataFrame, getResidualsAsDataFrame
```
Comment (yannikschaelte) [inline python/amici/__init__.py]: looks reasonable, but hope it won't get more cases than clibs and hdf5 :)

File: python/amici/sbml_import.py [hunk 2]
```diff
@@ -1611,7 +1617,9 @@ def printWithException(self,math):
 
         """
         try:
-            return self.codeprinter.doprint(math)
+            ret = self.codeprinter.doprint(math)
+            ret = re.sub(r'(^|\W)M_PI(\W|$)', r'\1amici::pi\2', ret)
```
Comment (yannikschaelte) [inline python/amici/sbml_import.py]: why are there both M_PI and amici::pi again?

File: python/amici/sbml_import.py [hunk 3]
```diff
@@ -295,6 +295,7 @@ def sbml2amici(self,
                    sigmas=None,
                    verbose=False,
                    assume_pow_positivity=False,
+                   compiler=None
```
Comment (yannikschaelte) [inline python/amici/sbml_import.py]: required for windows ('mingw32'), just wonder if amici can memorize this at compile time

File: python/amici/setup.template.py [hunk 4]
```diff
@@ -25,22 +32,22 @@ def getAmiciLibs():
...
+h5pkgcfg = getHdf5Config()
+
+blaspkgcfg = getBlasConfig()
+linker_flags.extend(blaspkgcfg.get('extra_link_args', []))
+
+libraries = [*getAmiciLibs(), *blaspkgcfg['libraries']]
+if hdf5_enabled:
+    libraries.extend(['hdf5_hl_cpp', 'hdf5_hl', 'hdf5_cpp', 'hdf5'])
+
+sources = ['swig/TPL_MODELNAME.i', *getModelSources()]
```
Comment (yannikschaelte) [inline python/amici/setup.template.py]: honestly can't tell if all flags and options are set correctly, but looks reasonable

File: python/amici/setuptools.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    # try for hdf5 in standard locations
+    hdf5_include_dir_hints = ['/usr/include/hdf5/serial',
+                              '/usr/local/include',
+                              '/usr/include', # travis ubuntu xenial
+                              '/usr/local/Cellar/hdf5/1.10.2_1/include'] # travis macOS
+    hdf5_library_dir_hints = ['/usr/lib/x86_64-linux-gnu/', # travis ubuntu xenial
+                              '/usr/lib/x86_64-linux-gnu/hdf5/serial',
+                              '/usr/local/lib',
+                              '/usr/local/Cellar/hdf5/1.10.2_1/lib'] # travis macOS
+
```
Comment (yannikschaelte) [inline python/amici/setuptools.py]: standard for windows?

File: python/amici/sbml_import.py (hunk 2 above)
Comment (dweindl) [reply python/amici/sbml_import.py]: I did not find a way to convince CCodeprinter to directly use amici::pi. So here M_PI is replaced by amici::pi.

File: python/amici/sbml_import.py (hunk 3 above)
Comment (dweindl) [reply python/amici/sbml_import.py]: Could be convenient. Would have to create a config file in package directory. Feel free to open a separate issue on that.

File: python/amici/setuptools.py (hunk 5 above)
Comment (dweindl) [reply python/amici/setuptools.py]: Not sure there is anything remotely standard on Windows. If you have something, we can add it..

Comment (yannikschaelte) [reply python/amici/setuptools.py]: Well, the Program Files (x86) folders, but we can see that whence hdf5 for windows works.

File: python/amici/__init__.py (hunk 1 above)
Comment (dweindl) [reply python/amici/__init__.py]: Yes, that's a bit messy. The problem is mostly, that for testing it is very helpful if you can import the package without requiring the compiled extensions. And the second problem is that there is no good way yet for having hdf5 available everywhere.

**Pull request merged**